0000-miner-update.md

```diff
@@ -0,0 +1,199 @@
+looks like the target edge devices need to run their Yocto-based balenaOS to
+receive them.
+
+- What is the impact of not doing this?
+
+If we do not allow Hotspot owners to build and distribute their own custom miner
+software we eventually risk greater scrutiny by Hotspot operators, regulators
+and large cryptocurrency exchanges.
+
+# Unresolved Questions
+[unresolved]: #unresolved-questions
+
+- What parts of the design do you expect to resolve through the HIP process
+before this gets merged?
+
+What is the minimum viable product we can implement without sacrificing security
+or diminishing the user experience?
+
+- What parts of the design do you expect to resolve through the implementation
+of this feature?
+
+With the exception of image overlays all of the approaches being considered are
+new so the only way to assess them is through rapid prototyping.
+
+- What related issues do you consider out of scope for this HIP that could be
+addressed in the future independently of the solution that comes out of this HIP?
+
+How does a third party package and sign a miner release for download and
+verification by a consumer Hotspot?
+
+How do we facilitate the deployment of miner instances in the cloud that receive
+LoRa packets from third party edge radio gateways?
+
+# Deployment Impact
+[deployment-impact]: #deployment-impact
+
+- How will current users be impacted?
+
+End users will have the option to specify the URL of a miner package repository
+or image registry besides Helium's so that subsequent miner updates originate
+from that third party miner package repository or image registry. If no
+alternative URL is specified then the miner package repository or image registry
+utilized by a Hotspot defaults to Helium's. In either case miner updates continue
+to happen automatically so an end users should notice no difference unless they
+select an unreliable miner package or image supplier.
+
+- How will existing documentation/knowledgebase need to be supported?
+
+The final solution will need to include clear step-by-step instructions for
+miner package or image maintainers. This documentation will likely take the form
+of a GitHub project README for a Helium minerhaul repo that miner package
+maintainers will need to fork to deliver their updates.
+
+UI for entering the URL of an alternative miner package repository will need to
+be added to the Hotspot Settings page of the Helium mobile app. A related help
+article will also need to be added to the support section of Helium's website
+for end users.
+
+- Is this backwards compatible?
+
+We are not abandoning the Buildroot embedded Linux build system for existing
+first generation of Helium Hotspots so any miner update approach we adopt will
+need to integrate with Buildroot.
+
+# Success Metrics
+[success-metrics]: #success-metrics
+
+- What should we measure to prove a performance increase or decrease?
+
+We need measure the impact on blockchain throughput that running miner inside a
```

Comment:
This was my primary concern with the docker approach, although it does feel more modern/flexible.

Reply:
Do we use any specialized crypto processor instructions in miner? Are those preserved via Dockerization?

Reply:
We want to use ARM's NEON SIMD extension and ARMv8 crypto extensions. I believe we are already cross-compiling OpenSSL with the former but we are not yet cross-compiling RocksDB with the latter. Docker now supports multi-arch images including arm64 so I think those instructions are preserved.